clio-client/src/main/scala/org/broadinstitute/clio/client/webclient/ClioWebClient.scala

```diff
@@ -117,29 +126,31 @@ class ClioWebClient(
     }
   }
 
-  def getClioServerVersion: Future[HttpResponse] = {
+  def getClioServerVersion: Future[Json] = {
     dispatchRequest(HttpRequest(uri = "/version"))
+      .flatMap(unmarshal[Json])
   }
 
-  def getClioServerHealth: Future[HttpResponse] = {
+  def getClioServerHealth: Future[Json] = {
     dispatchRequest(HttpRequest(uri = "/health"))
+      .flatMap(unmarshal[Json])
   }
 
   def getSchema(
     transferIndex: TransferIndex
-  )(implicit credentials: HttpCredentials): Future[HttpResponse] = {
+  )(implicit credentials: HttpCredentials): Future[Json] = {
     dispatchRequest(
       HttpRequest(uri = s"/api/v1/${transferIndex.urlSegment}/schema")
         .addHeader(Authorization(credentials = credentials))
-    )
+    ).flatMap(unmarshal[Json])
   }
 
-  def upsert[M <: TransferMetadata[M]](
-    transferIndex: TransferIndex,
-    key: TransferKey,
-    metadata: M
-  )(implicit credentials: HttpCredentials,
-    encoder: Encoder[M]): Future[HttpResponse] = {
+  def upsert[TI <: TransferIndex](transferIndex: TI)(
```

Comment:
While I was in here tweaking things, I changed these signatures to lock things down a bit more, like we've done in the `*Executor`.